nixos/modules/services/misc/transfer-sh.nix

```diff
@@ -0,0 +1,99 @@
+{ config, lib, pkgs, ... }:
+
+let
+  cfg = config.services.transfer-sh;
+  inherit (lib)
+    types optional mkDefault mkEnableOption mkPackageOption
+    mkIf mkOption mapAttrs isBool getExe boolToString mdDoc;
+in
+{
+  options.services.transfer-sh = {
+    enable = mkEnableOption (mdDoc "Easy and fast file sharing from the command-line");
+
+    package = mkPackageOption pkgs "transfer-sh" { };
+
+    settings = mkOption {
+      type = types.submodule { freeformType = with types; attrsOf (oneOf [ bool int str ]); };
+      default = { };
+      example = {
+        LISTENER = ":8080";
+        BASEDIR = "/var/lib/transfer.sh";
+        TLS_LISTENER_ONLY = false;
+      };
+      description = mdDoc ''
+        Additional configuration for transfer-sh, see
+        <https://github.com/dutchcoders/transfer.sh#usage-1>
+        for supported values.
+
+        For secrets use secretFile option instead.
+      '';
+    };
+
+    provider = mkOption {
+      type = types.enum [ "local" "s3" "storj" "gdrive" ];
+      default = "local";
+      description = mdDoc "Storage providers to use";
+    };
+
+    secretFile = mkOption {
+      type = types.nullOr types.path;
+      default = null;
+      example = "/run/secrets/transfer-sh.env";
+      description = mdDoc ''
+        Path to file containing environment variables.
+        Useful for passing down secrets.
+        Some variables that can be considered secrets are:
+         - AWS_ACCESS_KEY
+         - AWS_ACCESS_KEY
+         - TLS_PRIVATE_KEY
+         - HTTP_AUTH_HTPASSWD
```

Comment:
```suggestion
         - HTTP_AUTH_HTPASSWD
         - GA_KEY
         - HTTP_AUTH_PASS
         - USERVOICE_KEY
         - AWS_SECRET_KEY
         - STORJ_ACCESS

```

Reply:
In case you want to add a explicit warning about these, it can be done like this:

https://github.com/pinpox/nixos/blob/daf4adb3a91def671be765dfd10e9100b51b71c4/modules/transfer-sh/default.nix#L450-L466

Reply:
Since secretFile has been used, there is no danger or need for a warning.
I used `Some variables` so I didn't list them all. That would be verbose.

Reply:
>Since secretFile has been used, there is no danger or need for a warning.

Ok. I wanted to prevent users from putting these in the normal `settings` as an extra precaution, but I guess it's not strictly necessary.